# HG changeset patch
# User Torsten <mttl@mailbox.org>
# Date 1548537438 -3600
#      Sat Jan 26 22:17:18 2019 +0100
# Branch stable
# Node ID 931ace437d5d1cb94c9e7d20c595913d60c2c1b2
# Parent  48ddf51a6ef7383a29800b621f153e2e91b2cc83
fix possible memleak by avoiding bare poointer (bug #55440)

* shortcut-manager.cc (do_import_export): do not use a pointer for the
  QSettings object that is used for importing or esporting custom shortcuts

diff --git a/libgui/src/shortcut-manager.cc b/libgui/src/shortcut-manager.cc
--- a/libgui/src/shortcut-manager.cc
+++ b/libgui/src/shortcut-manager.cc
@@ -688,30 +688,30 @@ namespace octave
           file = QFileDialog::getSaveFileName (this,
                                                tr ("Export shortcuts to file..."), QString (),
                                                tr ("Octave Shortcut Files (*.osc);;All Files (*)"),
                                                nullptr, QFileDialog::Option (opts));
 
         if (file.isEmpty ())
           return false;
 
-        QSettings *osc_settings = new QSettings (file, QSettings::IniFormat);
+        QSettings osc_settings (file, QSettings::IniFormat);
 
-        if (! osc_settings)
+        if (osc_settings.status () !=  QSettings::NoError)
           {
             qWarning () << tr ("Failed to open %1 as Octave shortcut file")
                         .arg (file);
             return false;
           }
         else
           {
             if (action == OSC_IMPORT)
-              import_shortcuts (osc_settings);   // import (special action)
+              import_shortcuts (&osc_settings);   // import (special action)
             else if (action == OSC_EXPORT)
-              do_write_shortcuts (osc_settings, false); // export, (save settings)
+              do_write_shortcuts (&osc_settings, false); // export, (save settings)
           }
       }
     else
       {
         import_shortcuts (nullptr);
       }
 
     return true;
